gateway/mw_auth_key.go

```diff
@@ -95,7 +97,7 @@ func (k *AuthKey) ProcessRequest(_ http.ResponseWriter, r *http.Request, _ inter
 	}
 
 	// skip auth key check if the request is looped.
-	if ses := ctxGetSession(r); ses != nil && !ctxCheckLimits(r) {
+	if ses := ctxGetSession(r); ses != nil && httpctx.IsSelfLooping(r) {
```

Comment:
```suggestion
	if ses := ctxGetSession(r); ses != nil && !ctxCheckLimits(r) {
```

ctxCheckLimits already invokes IsSelfLooping (unnecessary line change)

Reply:
this is causing some other flows to break, hence decided to go with SelfLooping check

Reply:
```
api.go:func ctxCheckLimits(r *http.Request) bool {
grep: gateway.test: binary file matches
mw_api_rate_limit.go:	if !ctxCheckLimits(r) {
mw_organisation_activity.go:	if !ctxCheckLimits(r) {
mw_rate_limiting.go:	if !ctxCheckLimits(r) {
```

Is there a particular understanding as to why we had used ctxCheckLimits in auth? No unit test for the function, and ctx.CheckLoopLimits key from the context returns false by default (unset).